api-compact.yaml

```diff
@@ -85,9 +91,11 @@ x-amazon-apigateway-integration: &apigateway_integration
   type: "http"
   requestParameters:
     integration.request.header.Authorization: "method.request.header.Authorization"
+    integration.request.header.Host: "${stageVariables.enterprise_host}"
```

Comment:
I think it would be best to add new stage variables (e.g enterprise_api_host, discovery_api_host) that contain api gateway's domain. We can then use these varaibles in the configuration. 

@efagin , @mattdrayer  Any thoughts on this approach ?